test/test_permeation_problem.py

```diff
@@ -96,23 +91,39 @@ def siverts_law(T, S_0, E_S, pressure):
 
         my_model.solver.solve(u)
 
-        # post process
-        surface_flux = form(my_model.D * dot(grad(u), n) * ds(2))
+        mobile_xdmf.write_function(u, t)
+
+        surface_flux = form(my_model.D * dot(grad(u), n) * my_model.ds(2))
         flux = assemble_scalar(surface_flux)
         flux_values.append(flux)
         times.append(t)
 
-        # export
-        np.savetxt("outgassing_flux.txt", np.array(flux_values))
-        np.savetxt("times.txt", np.array(times))
-
-        mobile_xdmf.write_function(u, t)
-
-        # update previous solution
         mobile_H.prev_solution.x.array[:] = u.x.array[:]
 
     mobile_xdmf.close()
 
+    # analytical solution
+    S = S_0 * exp(-E_S / F.k_B / float(temperature))
+    permeability = float(my_model.D) * S
+    times = np.array(times)
+
+    n_array = np.arange(1, 10000)[:, np.newaxis]
+    summation = np.sum(
+        (-1) ** n_array
+        * np.exp(-((np.pi * n_array) ** 2) * float(my_model.D) / L**2 * times),
+        axis=0,
+    )
+    analytical_flux = P_up**0.5 * permeability / L * (2 * summation + 1)
+
+    flux_values = np.array(np.abs(flux_values))
+    normalised_analytical_flux = analytical_flux / analytical_flux.max()
+    normalised_compute_flux = flux_values / flux_values.max()
+
+    diff = np.abs(normalised_analytical_flux - normalised_compute_flux)
+    error = diff.mean()
+
+    assert error < 0.001
```

Comment:
This is the absolute mean difference. We may want to go for the relative difference instead.

Imagine a case where the computed flux is zero at all times, and the analytical flux is always below 0.001 H/m2/s

Then the absolute difference will be below 0.001 (because 0.001 - 0 = 0.001) and yet the code doesn't work!